kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java

```diff
@@ -29,4 +54,18 @@ public static String getStoreTypeOrPlatformDefault(String storeType) {
     public boolean definesKey() {
         return key != null;
     }
+
+    @Override
+    public TlsClientAuth clientAuth() {
+        // We can't just default this in the canonical constructor as it then becomes a nonsensical property of Upstream TLS config
+        return clientAuth == null ? TlsClientAuth.NONE : clientAuth;
+    }
+
+    @VisibleForTesting
+    boolean validateClientAuth() {
```

Comment:
At first glance, I assumed the return value was whether client auth was being enforced or not.    It is not, it whether the object is in a validate state.  Which is always true, otherwise the object would fail construction.

I'm not really appreciating the value of exposing this method for test.  I'd just inline it.

Reply:
Initially I had the validation independent of clothe ctor and thus had tests around it. When I moved it into the ctor I took the lazy option of keeping the method rather than rethinking how to assert the behaviour.